File: leetcode/solutions/0540.SingleElementInASortedArray/singleNonDuplicate_test.go
package solution

import (
	"testing"
)

func Test_0540_SingleElementInASortedArray(t *testing.T) {
	t.Parallel()

	tests := []struct {
		in  []int
		out int
	}{
		{
			[]int{1, 1, 2, 3, 3, 4, 4, 8, 8},
			2,
		},
		{
			[]int{3, 3, 7, 7, 10, 11, 11},
			10,
		},
	}

	for _, tt := range tests {
		tt := tt

		t.Run("Test", func(t *testing.T) {
			t.Parallel()

			got := singleNonDuplicate(tt.in)

			if got != tt.out {
				t.Errorf("got %#v want %#v", got, tt.out)
			}
		})
	}
}

File: leetcode/solutions/0540.SingleElementInASortedArray/singleNonDuplicate_test.ts
import { assertEquals } from "https://deno.land/std/testing/asserts.ts";
import singleNonDuplicate from "./singleNonDuplicate.ts";

Deno.test("0540. Single Element in a Sorted Array", () => {
  type test = [number[], number];
  const tests: test[] = [
    [[1, 1, 2, 3, 3, 4, 4, 8, 8], 2],
    [[3, 3, 7, 7, 10, 11, 11], 10],
  ];

  for (const [arg, want] of tests) {
    assertEquals(singleNonDuplicate(arg), want);
  }
});
